eth/eventsyncer/event_syncer.go

```diff
@@ -80,6 +85,21 @@ func (es *EventSyncer) Healthy(ctx context.Context) error {
 	if time.Since(es.lastProcessedBlockChange) > es.stalenessThreshold {
 		return fmt.Errorf("syncing is stuck at block %d", lastProcessedBlock.Uint64())
 	}
+
+	return es.blockBelowThreshold(ctx, lastProcessedBlock)
+}
+
+func (es *EventSyncer) blockBelowThreshold(ctx context.Context, block *big.Int) error {
+	header, err := es.executionClient.HeaderByNumber(ctx, block)
+	if err != nil {
+		return fmt.Errorf("failed to get header for block %d: %w", block, err)
+	}
+
+	// #nosec G115
+	if header.Time > 0 && header.Time < uint64(time.Now().Add(-es.stalenessThreshold).Unix()) {
```

Comment:
```suggestion
	if time.Since(time.Unix(header.Time, 0)) < es.stalenessThreshold) {
```

Reply:
> we cant ignore zero time because it may also happen because of an Eth client bug (we've actually witnessed a similar bug before with another field)
> do we create the blocks in the simulator? if so we can probably control the time (ideally)
> if we have no choice, we could set stalenessThreshold to MaxUint64 in only the failing tests